helpers/synccounter.go

```diff
@@ -0,0 +1,31 @@
+package helpers
+
+import "sync"
+
+type SyncCounter struct {
+	c  int
+	mu sync.Mutex
```

Comment:
I think https://pkg.go.dev/sync/atomic#Int64 is the leanest way to implement a this.